warehouse/templates/packaging/detail.html

```diff
@@ -244,11 +244,9 @@ <h2 class="no-top-padding">Reverse Dependencies (TODO)</h2>
             <div class="release{{ ' release--latest' if loop.first else ' release--oldest' if loop.last }}{{ ' release--current' if is_current }}">
               <div class="release__meta">
                 {% if is_current %}
-                <p class="release__version">{{ hrelease.version }}</p>
                 <span class="badge">This version</span>
-                {% else %}
-                <p class="release__version"><a href="{{ request.route_path('packaging.release', name=release.project.name, version=hrelease.version) }}">{{ hrelease.version }}</a></p>
                 {% endif %}
+                <p class="release__version"><a href="{{ request.route_path('packaging.release', name=release.project.name, version=hrelease.version) }}">{{ hrelease.version }}</a></p>
```

Comment:
This would put the "THIS VERSION" badge _before_ the link, but it should be after.